src/parser/jobs/ast/Lightspeed.js

```diff
@@ -0,0 +1,177 @@
+		if (this._active) { return }
+		this._startLightspeed(event.timestamp)
+	}
+
+	_onRefreshLightspeed() {
+		if (!this._active) { return }
+		this._lightspeed.extended = true
+	}
+
+	_onRemoveLightspeed() {
+		this._stopAndSave()
+	}
+
+	_onComplete() {
+		// Clean up any existing casts
+		if (this._active) {
+			this._stopAndSave()
+		}
+
+	}
+
+	_startLightspeed(start) {
+		this._active = true
+		this._lightspeed = {
+			start,
+			end: null,
+			extended: false,
+			casts: [],
+		}
+
+		this._ctIndex = this.castTime.set('all', LIGHTSPEED_CAST_TIME_MOD)
+	}
+
+	_stopAndSave(endTime = this.parser.currentTimestamp) {
+		// Make sure we've not already stopped this one
+		if (!this._active) {
+			return
+		}
+
+		this._active = false
+		this._lightspeed.end = endTime
+
+		this._history.push(this._lightspeed)
+
+		this.castTime.reset(this._ctIndex)
+
+	}
+
+	activeAt(time) {
+		// If it's during the current one, easy way out
+		if (this._active && this._lightspeed.start <= time) {
+			return true
+		}
+
+		return this._history.some(lightspeed => lightspeed.start <= time && lightspeed.end >= time)
+	}
+
+	output() {
+		const noCastsMessage = <Fragment>
+			<p>
+				<span className="text-error"><Trans id="ast.lightspeed.messages.no-casts">Zero casts recorded for <ActionLink {...ACTIONS.LIGHTSPEED} /></Trans></span>
+			</p>
+		</Fragment>
+
+		const panels = this._history.map(lightspeed => {
+			const numGcds = lightspeed.casts.filter(cast => getAction(cast.ability.guid).onGcd).length
+			const mpSavings = lightspeed.casts
+				.filter(cast => getAction(cast.ability.guid).onGcd)
+				.reduce((totalSavings, cast) => getAction(cast.ability.guid).mpCost / 2 + totalSavings, 0)
+			// TODO: Use mpCostFactor instead of mpCost to be level agnostic
+
+			return {
+				key: lightspeed.start,
+				title: {
+					content: <>
+						{this.parser.formatTimestamp(lightspeed.start)}
+						&nbsp;-&nbsp;{numGcds} GCDs
+						&nbsp;-&nbsp;{mpSavings} MP saved
+						{lightspeed.extended && <span className="text-info">&nbsp;(extended)</span>}
+					</>,
+				},
+				content: {
+					content: <Rotation events={lightspeed.casts}/>,
+				},
+			}
+		})
+
+		const lightspeedDisplay = <Accordion
+			exclusive={false}
+			panels={panels}
+			styled
+			fluid
+		/>
+
+		return <Fragment>
+			<p>
+				<Trans id="ast.lightspeed.messages.explanation">
+				Some of the applications of <ActionLink {...ACTIONS.LIGHTSPEED} /> include MP savings on heavy healing segments, keeping casts up while on the move and for weaving OGCDs.
+				To further complicate usage, <ActionLink {...ACTIONS.ESSENTIAL_DIGNITY} /> can reduce the cooldown, and the buff can be extended by <ActionLink {...ACTIONS.CELESTIAL_OPPOSITION} />.<br/><br/>
+				At this point of time it's difficult to identify what is optimal, since each fight calls for a different strategy.
+				</Trans>
+			</p>
+			{panels.length === 0 && noCastsMessage}
```

Comment:
I'd rather the output just hide itself completely if there were no casts. Not using lightspeed is possibly a suggestion instead?